src/blr/cli/tagfastq.py

```diff
@@ -0,0 +1,132 @@
+    corrected canonical sequence
+    :param open_file: starcode tabular output file.
+    :return: dict: raw sequences pointing to a namedtuple containing corrected canonical sequence and id.
+    """
+    target = namedtuple("Cluster", ['seq', 'id'])
+    corrected_barcodes = dict()
+    for id, cluster in tqdm(enumerate(open_file.readlines(), start=1), desc="Corrected barcodes processed"):
+        canonical_seq, _, cluster_seqs = cluster.strip().split("\t", maxsplit=3)
+        cluster_target = target(seq=canonical_seq, id=id)
+        corrected_barcodes.update({raw_seq: cluster_target for raw_seq in cluster_seqs.split(",")})
+    return corrected_barcodes
+
+
+def parser_raw_barcodes(open_file):
+    """
+    Parse FASTA/FASTQ containing barcode sequences and return a dictionary with entry headers pointing to a
+    raw barcodes sequence.
+    :param open_file: dnaio odject.
+    :return: dict: entry headers pointing to a raw barcodes sequence
+    """
+    header_barcodes = dict()
+    for barcode in tqdm(open_file, desc="Raw barcodes processed"):
+        header, _ = barcode.name.split(maxsplit=1)
+        header_barcodes[header] = barcode.sequence
+    return header_barcodes
+
+
+def add_arguments(parser):
+    parser.add_argument(
+        "input1", metavar='<INPUT1>',
+        help="Input FASTQ/FASTA file. Assumes to contain read1 if given with second input file. If only "
+             "input1 is given, input is assumed to be an interleaved. If reading from stdin"
+             "is requested use '-' as a placeholder.")
+    parser.add_argument(
+        "input2", nargs='?', metavar='<INPUT2>',
+        help="Input  FASTQ/FASTA for read2 for paired-end read. Leave empty if using interleaved.")
+    parser.add_argument(
+        "-o1", "--output1", default=None, metavar='<OUTPUT1>',
+        help="Output FASTQ/FASTA file name for read1. If not specified the result is written to stdout"
+             " as interleaved. If output1 given but not output2, output will be written as "
+             "interleaved to output1.")
+    parser.add_argument(
+        "-o2", "--output2", default=None, metavar='<OUTPUT2>',
+        help="Output FASTQ/FASTA name for read2. If not specified but -o1 given the result is written"
+             " as interleaved to o1.")
+    parser.add_argument(
+        "-rb", "--barcodes-raw", required=True, metavar='<RAW BARCODES>',
+        help="FASTQ/FASTA for raw barcodes.")
+    parser.add_argument(
+        "-cb", "--barcodes-corrected", required=True, metavar='<CORRECTED BARCODES>',
+        help="FASTQ/FASTA for error corrected barcodes. Currently accepts output from starcode clustering with "
+             "'--print-clusters' enabled.")
+    parser.add_argument(
+        "-bc", "--barcode-cluster-tag", metavar="<STRING>", type=str, default="BX",
+        help="BAM file tag where barcode cluster id is stored. 10x genomics longranger output "
+             "uses 'BX' for their error corrected barcodes. DEFAULT: BX")
```

Comment:
Don’t hard-code the default in the help message, but use `%(default)s` instead.

Reply:
In line with your previous comment about global configuration for argparse, maybe we could make argparse always add a `DEFAULT: %(default)s` string to help messages whenever `default`is not `None`?

Reply:
Sounds good in principle but there are too many exceptions when this wouldn’t work. I think this is something that needs to decided from option to option since it’s meant as human-readable text. For conciseness, for example, I sometimes do not write out the default if it is clear from the context. For example, if `--reverse` means "Print foo file in reverse", it’s obvious that the default is to not print the file in reverse. And whether I have encode this as `None` and `True` or `forward` and `reverse` in the value is independent of this.

(By the awy, I suggest to use normal casing: `Default: %(default)s`.)

Reply:
Though I guess that specific case would be solved by having a convention to use the `"store_true"` flag when applicable (for all "binary" options). Either way since this was just one example I'll trust your experience regarding this and leave it up for future me to revisit this if I feel it is necessary.

Reply:
I don’t want to argue from authority, so I’ve looked through the options in Cutadapt. It would work there for seven of the options if you extend the rules to not print the default when `default=''` or `default=[]` (for `action="append"`). On the other hand, there are a couple of options for which I print out a default even though `default=None`, such as "Default: no limit" (for maximum read length) or "Default: output to same file as trimmed reads" (for `--untrimmed-output`).

If at some point this is the only thing in the code that prevents it from being beautiful, I’d so go for it.

Reply:
Thanks for taking the time to look it up! 

Yeah, that makes sense that some options could require a more readable or non-programming term to explain defaults to a user.